drift.py

```diff
@@ -1,146 +1,90 @@
+import json
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from datetime import datetime
+from subprocess import run as subprocess_run
+
+import requests
 from helicase import Helicase
-from datetime import *
-import argparse
-import json, requests
-import subprocess, sys
+from spack import *
```

Comment:
Don't use `*` imports -- import what you need.

In this case I think you want `from spack.spec import Spec`

Reply:
Hahaha @vsoch and I were just having this conversation and I think we figured we'd go with `from spack import *` because it's done everywhere in spack's own codebase. I'm very happy to fix this up however!

Reply:
Only in packages -- not in core.  It is actually a no-op in packages for import speed reasons.  There is actually nothing *in* the `spack` namespace except `spack.spack_version_info` and `spack_version`.

Other magic handles inserting everything in `spack.pkgkit` into the packages -- spack packages are actually imported by their own custom importer.  So `hdf5/package.py` from the `builtin` repo comes in as the `spack.pkg.builtin.hdf5` module.

The reason we did the refactor is b/c spack startup time got really slow, and mostly it was because literally everything was being dragged in when you imported the `spack` namespace.  Imports in python are horribly slow and in general I recommend having absolutely nothing, or close to it, in your `__init__.py` files because one day you'll want to change it and won't be able to.

Reply:
Hmm that's not what I said, I said module imports like `import spack.spec` and then use `spack.spec.Spec`. I agree with Todd imports with * are not good practice.

Reply:
Sorry I must have been confusing!

Reply:
Haha, but I totally remember editing my first spack package and seeing `from spack import *` and then _what is this witchcraft?!_ :sparkles:

Reply:
yeah sorry about all the witchcraft 🧙💥 https://github.com/spack/spack/blob/develop/lib/spack/spack/util/imp/importlib_importer.py

For real crimes, see `imp_importer` right next to it.